### PR TITLE
Debug info for unsupported constraints

### PR DESCRIPTION
**Pull request by yashcodes**
Closes [#2143](https://github.com/jump-dev/JuMP.jl/issues/2143)

### CODE REVIEW

**Comment by yashcodes** [comment]
Right now it just prints out all the information of Unsupported constraints (NOT giving any error). I cant really figure put how can we put all this information in Error.

**Comment by yashcodes** [comment]
This is supposed to work but I dont know why the julia session quit whenever I try to access function `debug_string`

File: src/constraints.jl [hunk 1]
```diff
@@ -376,13 +376,20 @@ function check_belongs_to_model(con::VectorConstraint, model)
     end
 end
 
+function debug_string(b::MOIB.LazyBridgeOptimizer, F::Type{<:MOI.AbstractFunction}, S::Type{<:MOI.AbstractSet})
+    io = IOBuffer()
+    MOIB.debug(b, F, S; io = io)
+    return String(resize!(io.data, io.size))
```
Comment (odow) [inline src/constraints.jl]:
```Julia
sprint(io -> MOI.Bridges.debug(b, F, S; io = io))
```

**Comment by yashcodes** [comment]
@odow As you suggested I tried using `sprint` but after that I got `StackOverflowError:` by infinite recursive calling of function `add_unsupported` when I tried [test](https://github.com/JuliaOpt/JuMP.jl/pull/2203/files#diff-2597e8531a21ace55ac1201c236cf47cL251)

**Comment by odow** [comment]
I'm not sure where the stack overflow comes. Perhaps you could try isolating the issue. Can you reproduce it outside this PR just by calling `MOI.Bridges.debug` with some arguments?

**Comment by yashcodes** [comment]
I tried calling `sprint(io -> MOI.Bridges.debug(bridged, F, S; io = io))` with same arguments with which it was called inside ` moi_add_constraint`.
Its really weird, Julia session ends unexpectedly.
`
 julia> using JuMP
[ Info: Precompiling JuMP [4076af6c-e467-56ae-b986-b466b2749572]

julia> function mock_factory()
                       mock = MOIU.MockOptimizer(MOIU.Model{Float64}(),
                                                 eval_variable_constraint_dual=false)
                       optimize!(mock) = MOIU.mock_optimize!(mock, [1.0],
                               (MOI.SingleVariable, MOI.GreaterThan{Float64}) => [2.0])
                       MOIU.set_mock_optimize!(mock, optimize!)
                       return mock
                   end
mock_factory (generic function with 1 method)

julia> model = Model(mock_factory)
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: EMPTY_OPTIMIZER
Solver name: Mock

julia> @variable(model, x)
x

julia> F = MOI.SingleVariable
MathOptInterface.SingleVariable

julia> include("/Users/yash/.julia/dev/JuMP/test/nonnegative_bridge.jl")

julia> S = Nonnegative
Nonnegative

julia> sprint(io -> MOI.Bridges.debug(backend(model).optimizer, F, S; io = io))

Julia has exited. Press Enter to start a new session.
`

**Comment by yashcodes** [comment]
Tests are failing because its using outdated version of MathOptInterface, Tests will run successfully once the new version is released.

**Pull request merged**